core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -2204,10 +2219,10 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
 
   /** The "DATE_TRUNC(date, timeUnit)" function (BigQuery);
    * truncates a DATE value to the beginning of a timeUnit. */
-  @LibraryOperator(libraries = {BIG_QUERY, SPARK})
+  @LibraryOperator(libraries = {BIG_QUERY, SPARK, POSTGRESQL})
   public static final SqlFunction DATE_TRUNC =
       SqlBasicFunction.create("DATE_TRUNC",
-          ReturnTypes.ARG0_NULLABLE,
+          ReturnTypes.ARG0_EXCEPT_STRING_NULLABLE,
```

Comment:
we have changed this type for BigQuery and spark dialects too. Did we verify whether this is as expected for those dialects too?

Reply:
Yeah @divyanshuS27  we verified it.Actually in spark and postgres function syntax is DATE_TRUNC(string, datetime_types) so the previous implementation returns the first argument type so if it is string its returning string type.We are modifying into datetime types.